prov/tcp/src/xnet_cm.c

```diff
@@ -63,7 +63,7 @@ xnet_recv_cm_msg(SOCKET sock, struct xnet_cm_msg *msg)
 			"Failed to read cm header, ret: %zd, sockerr: %d\n",
 			ret, ofi_sockerr());
 		msg->hdr.seg_size = 0;
-		return ofi_sockerr() ? -ofi_sockerr() : -FI_EIO;
+		return ret == 0 ? -FI_EIO : -ofi_sockerr();
```

Comment:
`0 < ret < len` is also not an error condition.

Reply:
You can change the check to: ```ret >= 0 ? -FI_EIO ...```.  CM messages are small and the first data exchanged on the socket.  The code expects it to be delivered whole (doesn't support partial delivery of the CM message).  The tcp stack allocates at least an mtu size buffer for the socket, so the chances of 0 < ret < len occurring is zero in practice.

Reply:
Yeah I would think so, but I'm convinced by the comment:
```
/* The underlying socket has the POLLIN event set.  The entire
 * CM message should be readable, as it fits within a single MTU
 * and is the first data transferred over the socket.
 */
 ```
@shefty yeah I can make that change.